src/v/model/transform.h

```diff
@@ -71,4 +73,61 @@ struct transform_metadata
           name, input_topic, output_topics, environment, uuid, source_ptr);
     }
 };
+
+/**
+ * A (possibly inconsistent) snapshot of a transform.
+ *
+ * We capture the metadata associated with the transform for as well as the
+ * state for each processor.
+ */
+struct transform_report
+  : serde::
+      envelope<transform_report, serde::version<0>, serde::compat_version<0>> {
+    // The overall metadata for a transform.
+    transform_metadata metadata;
+
+    struct processor
+      : serde::
+          envelope<processor, serde::version<0>, serde::compat_version<0>> {
+        enum class state : uint8_t { unknown, inactive, running, errored };
+        model::partition_id id;
+        state status;
+        model::node_id node;
+        ss::shard_id core;
```

Comment:
I think it is ok if its useful for reporting and that sort of thing. I just wanted to make sure it wasn't going to ever be used for any sort of scheduling / affinity etc... because we are planning to remove the static mapping of partitions to cores in the near future where a partition can more easily migrate to other cores on a node.